src/app/collective-rewards/user/hooks/useGetBuildersByState.ts

```diff
@@ -17,7 +15,10 @@ export const useGetBuildersByState = <T extends Builder>(
 
       if (!stateFlags) return true
 
-      return stateFlags.some(value => builderStateFlags[value])
+      return Object.keys(stateFlags).every(key => {
```

Comment:
By using `every` we do expect the function to be called with that filter that works with & conditions (all of the conditions must be met). Is it the way we use this function in other places also?

Reply:
yes I update it, now what it does is that you send the object that you want to return, I may introduce some unit tests if you want.

for example if I send:
```
{
    activated: true,
    communityApproved: true,
    kycApproved: true,
    paused: false,
  }
```

It will return all the builders that have those specific flags, since Im not taking into account the revoked, it just ignore it.